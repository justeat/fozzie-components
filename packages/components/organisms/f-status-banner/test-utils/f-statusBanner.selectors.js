export const COMPONENT = '[data-test-id="main-banner-container"]';
export const ADDRESS_BOX = '[data-test-id="address-box-input"]';
export const BUTTON = '[data-test-id="find-restaurants-button"]';
