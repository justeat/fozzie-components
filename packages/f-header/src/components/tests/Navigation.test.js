import { shallowMount } from '@vue/test-utils';
import Navigation from '../Navigation.vue';

describe('Navigation', () => {
    it('should be defined', () => {
        const propsData = {
            userInfo: {},
            accountLogin: {},
            accountLogout: {},
            navLinks: {},
            openMenuText: 'Open menu',
            deliveryEnquiry: {},
            showDeliveryEnquiry: false
        };
        const wrapper = shallowMount(Navigation, { propsData });
        expect(wrapper.exists()).toBe(true);
    });

    it('should show "Deliver for Just Eat" in nav if prop true and content there', () => {
        // Arrange
        const propsData = {
            navLinks: {},
            deliveryEnquiry: {
                text: 'Deliver with Just Eat',
                url: 'https://couriers.just-eat.co.uk/application?utm_medium=referrer&utm_source=just-eat.co.uk&utm_campaign=ex1140-header',
                gtm: 'click_delivery_job'
            },
            showDeliveryEnquiry: true
        };

        // Act
        const wrapper = shallowMount(Navigation, { propsData });

        // Assert
        expect(wrapper.find('[data-js-test="delivery-enquiry"]').exists()).toBe(true);
    });

    it('should NOT show "Deliver for Just Eat" in nav if prop true and content there', () => {
        // Arrange
        const propsData = {
            navLinks: {},
            deliveryEnquiry: {},
            showDeliveryEnquiry: false
        };

        // Act
        const wrapper = shallowMount(Navigation, { propsData });

        // Assert
        expect(wrapper.find('[data-js-test="delivery-enquiry"]').exists()).toBe(false);
    });

    it('should show "navLinks" and "logout" if the user is logged in and HAS nav link data', () => {
        // Arrange
        const propsData = {
            navLinks: {
                orderHistory: {
                    text: 'Your orders',
                    url: '/order-history',
                    gtm: 'click_account_your_orders'
                },
                savedCards: {
                    text: 'Your saved cards',
                    url: '/member/savedcards',
                    gtm: 'click_account_saved_cards'
                }
            },
            userInfo: {
                isAuthenticated: true
            }
        };

        // Act
        const wrapper = shallowMount(Navigation, { propsData });

        // Assert
        expect(wrapper.find('[data-js-test="logout"]').exists()).toBe(true);
        expect(wrapper.find('[data-js-test="nav-links"]').exists()).toBe(true);
    });

    it('should NOT show "navLinks" but show "logout" if the user is logged in but DOES NOT have nav link data', () => {
        // Arrange
        const propsData = {
            navLinks: {},
            userInfo: {
                isAuthenticated: true
            }
        };

        // Act
        const wrapper = shallowMount(Navigation, { propsData });

        // Assert
        expect(wrapper.find('[data-js-test="logout"]').exists()).toBe(true);
        expect(wrapper.find('[data-js-test="nav-links"]').exists()).toBe(false);
    });

    it('should show "login if the user has no nav link data and is not authenticated"', () => {
        // Arrange
        const propsData = {
            navLinks: {},
            userInfo: {
                isAuthenticated: false
            }
        };

        // Act
        const wrapper = shallowMount(Navigation, { propsData });

        // Assert
        expect(wrapper.find('[data-js-test="login"]').exists()).toBe(true);
    });
});
