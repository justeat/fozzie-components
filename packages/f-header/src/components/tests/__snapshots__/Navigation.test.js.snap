// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render default component markup 1`] = `
<nav class="c-nav c-nav--global" istransparent="true"><button type="button" class="c-nav-trigger">

  </button> <input id="nav-trigger" type="checkbox" class="c-nav-trigger is-hidden"> <label for="nav-trigger" class="c-nav-toggle"><span class="c-nav-toggle-icon"></span></label>
  <div class="c-nav-container">
    <ul class="c-nav-list">
      <!---->
      <li class="c-nav-list-item has-sublist is-hidden"><a href="/" aria-expanded="false" aria-haspopup="true" class="c-nav-list-text">
          <profile-icon-stub class="c-nav-icon c-nav-icon--profile"></profile-icon-stub>
          <!---->
        </a>
        <ul class="c-nav-popoverList">
          <li class="c-nav-list-item c-nav-list-item--forceLast"><a tabindex="-1" data-trak="{
                                &quot;trakEvent&quot;: &quot;click&quot;,
                                &quot;category&quot;: &quot;engagement&quot;,
                                &quot;action&quot;: &quot;header&quot;,
                                &quot;label&quot;: &quot;undefined&quot;
                            }" class="c-nav-list-link">

            </a></li>
        </ul>
      </li>
      <li class="c-nav-list-item"><a rel="nofollow" data-trak="{
                        &quot;trakEvent&quot;: &quot;click&quot;,
                        &quot;category&quot;: &quot;engagement&quot;,
                        &quot;action&quot;: &quot;header&quot;,
                        &quot;label&quot;: &quot;undefined&quot;
                    }" class="c-nav-list-link">

        </a></li>
      <li class="c-nav-list-item c-nav-list-item--support"><a data-trak="{
                        &quot;trakEvent&quot;: &quot;click&quot;,
                        &quot;category&quot;: &quot;engagement&quot;,
                        &quot;action&quot;: &quot;header&quot;,
                        &quot;label&quot;: &quot;undefined&quot;
                    }" class="c-nav-list-link">

        </a></li>
    </ul>
  </div>
</nav>
`;
